src/interpreter/mod.rs

```diff
@@ -1371,6 +1375,13 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
         use syntax::ast::{IntTy, UintTy};
         let val = match (self.memory.pointer_size, &ty.sty) {
             (_, &ty::TyBool)              => PrimVal::Bool(self.memory.read_bool(ptr)?),
+            (_, &ty::TyChar)              => {
```

Comment:
It just occurred to me that `SwitchInt` should check for invalid `char` when switching over a `char` as well.